### PR TITLE
feat: sync script improvements ready for multi-page splitting

### PR DESCRIPTION
**Pull request by matteematt**
This is a change to the api sync plugin, this doesn't change any documentation files.

Features
1. Changes the docs manfiest strucutre to support file splitting of markdown files from FUI to multiple in the docs
2. Cleans up and recactors code in general

The splitting functionality has not yet been tested, and will be completed in a subsequent PR. This PR aims to just get all the necessary work in place to support that. Running this script should behave the same as it did before. To verify this
```
1. Check out this branch
2. Change the `foundation-layout` dependency to `14.144.0` in the main package.json
3. `npm i`
4. Comment out the line for `foundation-layout` in `plugins/api-docs/processedMap.js`
5. Run `npm run start:copy-docs`
6. There should be no changes to the layout files (readme/pictures/api docs - there is a change in the readmewhich is the removal of an extra space at the end of the line, as I fixed this when syncing the docs back upstream previously.
```

### CODE REVIEW

**Review by wjhendry — APPROVED** [review]

**Review by nick-genesis — APPROVED** [review]
Nice one!
Seems to work as described :)

image

File: plugins/api-docs/src/copyFunctions.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ *  etc.
+ *
+ * ...to render unlinked, so users actually see text like this instead:
+ *
+ * Implements: [NodeEnv](/web/filters/docs/api/foundation-filters.nodeenv)
+ *
+ * Replacing html tags like <b> with their markdown equivalent `**` fixes the issue, as the line starts with markdown.
+ */
+function cleanseMarkdownContent(input: string) {
+  return input.replace(/<!-- -->/g, "").replace(/<b>|<\/b>/g, "**");
```
Comment (nick-genesis) [inline plugins/api-docs/src/copyFunctions.ts:23]: Should we be doing this on the generation side?
Works fine for now :)

Comment (matteematt) [reply plugins/api-docs/src/copyFunctions.ts:23]: Yeah I think the fact that the docs are full of empty markdown comments is probably a bug with their docs generation tools 😂 there is an existing ticket somewhere to migrate to their newer version of the tool which can handle different OS line endings better too. 
But for now we'll have to leave it here.

**Review by MrBrunoWolff — APPROVED** [review]
image

Changes: ✅ 
Local test: ✅

**Pull request merged**